markdown/renderer.go

```diff
@@ -23,26 +23,54 @@ var (
 	codeBlockChars          = []byte("```")
 	tableHeaderColChar      = []byte{'-'}
 	tableHeaderAlignColChar = []byte{':'}
+	heading1UnderlineChar   = []byte{'='}
+	heading2UnderlineChar   = []byte{'-'}
 )
 
 // Ensure compatibility with Goldmark parser.
 var _ renderer.Renderer = &Renderer{}
 
 // Renderer allows to render markdown AST into markdown bytes in consistent format.
 // Render is reusable across Renders, it holds configuration only.
-type Renderer struct{}
+type Renderer struct {
+	Opts *renderer.Config
+}
 
 func (mr *Renderer) AddOptions(opts ...renderer.Option) {
-	c := renderer.Config{}
+	// this function is a bit of a bad design, as the options here are goldmark's HTML renderer
+	// structure... but goldmark just forwards it to us anyway
+	// (but it's goldmark's bad design, not ours :))
 	for _, o := range opts {
-		o.SetConfig(&c)
+		o.SetConfig(mr.Opts)
 	}
+}
 
-	// TODO(bwplotka): Add headers optionality (https://github.com/Kunde21/markdownfmt/issues/14).
+type UnderlineHeadings struct {
+	Enable bool
+}
+
+func (u UnderlineHeadings) SetConfig(config *renderer.Config) {
+	// same comment as in AddOptions
```

Comment:
Let's not add those comments everywhere, I think once it's enough -there is no plan to change it in future (: We are fitting Goldmark plan

Reply:
ok